File: HackerRank/Algorithms/Strings/String_construction.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function stringConstruction(s){
    // Complete this function
    var characters = new Array(26);
    var i = 0, copyCost = 0;
    var length = s.length;

    while (i < length) {
      var t = s.charCodeAt(i) - 'a'.charCodeAt(0);
      if (!characters[t]) {
        characters[t] = true;
        copyCost++;
      }
      i++;
    }

    return copyCost;
}

function main() {
    var q = parseInt(readLine());
    for(var a0 = 0; a0 < q; a0++){
        var s = readLine();
        var result = stringConstruction(s);
        process.stdout.write("" + result + "\n");
    }

}

File: HackerRank/Algorithms/Strings/String_construction.java
import java.util.Scanner;

public class Solution {

    static int stringConstruction(String s){
        // Complete this function
        boolean[] characters = new boolean[26];
        int i = 0, copyCost = 0;
        int length = s.length();

        while (i < length) {
          char t = s.charAt(i);
          if (!characters[t-'a']) {
            characters[t - 'a'] = true;
            copyCost++;
          }
          i++;
        }

        return copyCost;
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int q = in.nextInt();
        for(int a0 = 0; a0 < q; a0++){
            String s = in.next();
            int result = stringConstruction(s);
            System.out.println(result);
        }
    }
}
